### DISCUSSION TITLE
librdkafka vcpkg that includes Linux ARM64v8 and macOS ARM64v8 artifacts?

### DISCUSSION THREAD

**Discussion by jeremyosterhoudt** [microsoft/vcpkg#37609] [Q&A] [answered]
We are using the C++ version of RD Kafka and noticed we are unable to build for `ARM64` for `Linux` and `Mac` due to what looks like  `librdkafka` not being available in `ARM64` for `Linux` and `macOS` via vcpkg (maybe Windows as well but we didn't test).  The same build scripts are able to resolve `librdkafka` when used for x86/x64 on `Linux` and `Mac`.  Is there an easy way to confirm if they are indeed not published?  If they are not published is it a difficult task to add them to the distribution (happy to attempt a PR if someone can point me in the right direction).

**Comment by dg0yt** [comment, marked as answer]
~~~
$ vcpkg install librdkafka:arm64-linux
....
Elapsed time to handle librdkafka:arm64-linux: 21 s
Total install time: 26 s
The package librdkafka:arm64-linux provides CMake targets:

    find_package(RdKafka CONFIG REQUIRED)
    target_link_libraries(main PRIVATE RdKafka::rdkafka RdKafka::rdkafka++)
~~~

**Reply by jeremyosterhoudt** [reply]
Interesting, looks like I need to look at our build script a little closer.  Figured since it worked fine for x64 and I didn't see linux-arm in the Compatibility list using the package browser on vcpkg.io it may not be published, but looks like that assumption was bad.

Thank you for your time!

**Reply by dg0yt** [reply]
Don't look at vcpkg.io. It is always outdated. Sadly, no improvement in years.
Try https://vcpkg.link.

**Reply by dg0yt** [reply]
Or look at `ports/<PORT>/vcpkg.json`, field `supports`.
Or try `vcpkg install --dry-run <PORT>:triplet`.

But even if a platform is not directly excluded, it might still be recorded as CI baseline, or can be blocked by an unsupported dependency. (The `install --dry-run` must check the dependencies.)
